docs/firmware/golioth-firmware-sdk/light-db-stream/guide-lightdb-stream-object.md

```diff
@@ -0,0 +1,235 @@
+---
+id: by-object
+title: LightDB Stream (by object)
+sidebar_position: 2
+---
+
+Object-based LightDB functions can be used to send any number of time-series
+data values to the cloud with a single function call. A timestamp is
+automatically added to the data by the server and stored using a specific path
+that you provide. This data will accumulate along with all previously received
+values. If the chosen path does not already exist it will be created.
+
+Synchronous and asynchronous functions exist to set multiple data types and
+paths using your choice of these object types: `JSON` or `CBOR`.
+
+## Includes
+
+```c
+#include "golioth.h"
+```
+
+Including the `golioth.h` header file makes the Golioth API functions available
+to your program.
+
+## Preparing the Data
+
+The data you want to send must first be prepared as an object. This will include
+one or more key-values pairs. Choose between `JSON` or `CBOR` format.
+
+### JSON
+
+JSON payloads may be formatted using simple `snprintf()` style string
+substitution or a library like [cJSON](https://github.com/DaveGamble/cJSON).
+Here is a properly-formatted JSON example using all of the key-value pair
+examples from [the type-based stream examples](by-type#synchronous):
+
+```c
+/* Create a valid JSON string */
+char json_buf[] = "{\"name\":\"Golioth\",\"active\":true,\"data\":{\"reading\":1.337,\"calib\":42}}";
+```
+
+JSON is great for prototyping as it's human readable. When sending larger data
+sets, you may experience a bandwidth savings by using CBOR instead.
+
+### CBOR
+
+The same data may be formatted as a serialized CBOR object. On the device this
+should be done using a library of your choice. [The zCBOR
+library](https://github.com/NordicSemiconductor/zcbor) is already available as a
+submodule dependency of the Golioth Firmware SDK.
+
+```c
+/* Create serialized CBOR object */
+uint8_t cbor_buf[] = { 0xA3, 0x64, 0x6E, 0x61, 0x6D, 0x65, 0x67, 0x47, 0x6F, 0x6C, 0x69, 0x6F,
```

Comment:
It's not expected that users would craft CBOR messages by hand like they might with JSON. Should we include a snippet that shows how to use zcbor to create this message?

Reply:
Good point! I've added a commit that shows a CBOR encoding example.